src/picongpu/include/plugins/ChargeConservation.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2013-2015 Heiko Burau, Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "plugins/ILightweightPlugin.hpp"
+
+namespace picongpu
+{
+using namespace PMacc;
+
+namespace po = boost::program_options;
+
+/**
+ * @class ChargeConservation
+ * @brief maximum difference between electron charge density and div E
+ *
+ * WARNING: This plugin assumes a Yee-cell!
+ * Do not use it together with other field solvers like `directional splitting` or `Lehe`
+ */
+class ChargeConservation : public ILightweightPlugin
+{
+private:
+    std::string name;
+    std::string prefix;
+    uint32_t notifyPeriod;
+    std::ofstream* output_file;
```

Comment:
why are you using a pointer to ofstream instead of a direct instance of ofstream?